### PR TITLE
ROU-3748: Add client action to apply conditional format to the aggregate row

### PR DESCRIPTION
**Pull request by OS-giulianasilva**
This PR is for add a new client action to apply conditional format to the aggregate row.

### What was done
OutSystems.Providers.DataGrid.Wijmo.Feature:
- Fixed error not being throw for SetRowAsSelected

OutSystems.GridAPI.ConditionalFormat:
- Created SetNumberAggregateConditionalFormatting function

Providers.DataGrid.Wijmo.Features.ColumnAggregate:
- Created the _cellClasses Map to store the class to be applied to each column. The keys are the column binding and the element array os string containing the classNames to be added to the cell's aggregate.
- Implemented setConditionalFormat method:
	- Responsible for check if all the conditions to add the conditional format are respected.
	- Calls the addAggregateRules from Conditional Format
- Implemented addClass and removeClass methods:
	- Responsible for add and remove the cell aggregate classes from _cellClasses Map element associated with the column binding
- Implemented _formatItem method:
	- It is the formatItem event handler 
	- Responsible for add the classes stored in _cellClasses Map for each column aggregate everytime the view is refreshed.

Providers.DataGrid.Wijmo.Feature.ConditionExecuter
- Added the _isAggregate flag property
- Created the _addOrRemoveAggregateClass method, responsible to call the methods to add or remove classes from the column aggregates.
- Modified the execute handle the case of conditional format for columns aggregate 

Providers.DataGrid.Wijmo.Feature.ConditionalFormat
- Created the _mappedRulesAggregate Map to store the conditional format rules for each column aggregate.
- Created the _updateAggregateRows to handle classes update based on the conditional format for the aggregate rows
Created the addAggregateRules to add the conditional format rules to the binding from the aggregate rows.~

Providers.DataGrid.Wijmo.Feature:
- In setRowAsSelected method, fixed the error not being handled correctly

### Test Steps
1. Create a Grid with a number Column and ShowAggregate parameter set to true.
2. In the Grid OnInitialize event add the SetNumberAggregateConditionalFormatting
3. Set the mandatory parameter and the ConditionalFormat rule for the aggregate row
4. Check if the conditional format is applied to the aggregate row.

### Checklist
* [x] tested locally
* [x] documented the code
* [x] clean all warnings and errors of eslint
* [x] requires changes in OutSystems (if so, provide a module with changes)
* [ ] requires new sample page in OutSystems (if so, provide a module with changes)

### CODE REVIEW

File: src/Providers/DataGrid/Wijmo/Features/ColumnAggregate.ts [hunk 1]
```diff
@@ -40,5 +136,36 @@ namespace Providers.DataGrid.Wijmo.Feature {
...
             }
         }
+
+        /**
+         * Function to remove the aggregate cell class
+         *
+         * @param columnBinding {string} => The column binding of the aggregate to add the class
+         * @param className {string} => Classname to be added
+         */
+        public removeClass(columnBinding: string, className: string): void {
```
Comment (BenOsodrac) [inline src/Providers/DataGrid/Wijmo/Features/ColumnAggregate.ts:146]: @OS-giulianasilva lint warning, just about method order

**Review by BenOsodrac — APPROVED** [review]

File: src/Providers/DataGrid/Wijmo/Features/ColumnAggregate.ts (hunk 1 above)
Comment (OS-giulianasilva) [reply src/Providers/DataGrid/Wijmo/Features/ColumnAggregate.ts]: Fixed :)

**Review by BenOsodrac — DISMISSED** [review]

**Review by BenOsodrac — APPROVED** [review]

File: src/OSFramework/DataGrid/Feature/IColumnAggregate.ts [hunk 2]
```diff
@@ -1,5 +1,29 @@
...
+         * @param columnBinding {string} => The column binding of the aggregate to add the class
+         * @param className {string} => Classname to be added
+         */
+        addClass(columnBinding: string, className: string): void;
+
+        /**
+         * Function to remove the aggregate cell class
+         *
+         * @param columnBinding {string} => The column binding of the aggregate to add the class
+         * @param className {string} => Classname to be added
```
Comment (JoaoFerreira-FrontEnd) [inline src/OSFramework/DataGrid/Feature/IColumnAggregate.ts:17]: *to be removed

File: src/OSFramework/DataGrid/Feature/IColumnAggregate.ts [hunk 3]
```diff
@@ -1,5 +1,29 @@
...
+         */
+        addClass(columnBinding: string, className: string): void;
+
+        /**
+         * Function to remove the aggregate cell class
+         *
+         * @param columnBinding {string} => The column binding of the aggregate to add the class
```
Comment (JoaoFerreira-FrontEnd) [inline src/OSFramework/DataGrid/Feature/IColumnAggregate.ts:16]: *remove

File: src/OutSystems/GridAPI/ConditionalFormat.ts [hunk 4]
```diff
@@ -31,6 +31,49 @@ namespace OutSystems.GridAPI.ConditionalFormat {
         );
     }
 
+    /**
+     * Adds new conditional format rules to the desired aggregate binding.
+     *
+     * @export
+     * @param {string} gridID
+     * @param {string} binding Column binding
+     * @param {string} rules Rules for conditional formatting.
```
Comment (JoaoFerreira-FrontEnd) [inline src/OutSystems/GridAPI/ConditionalFormat.ts:40]: Review params above

File: src/OSFramework/DataGrid/Feature/IColumnAggregate.ts (hunk 3 above)
Comment (OS-giulianasilva) [reply src/OSFramework/DataGrid/Feature/IColumnAggregate.ts]: Fixed

File: src/OutSystems/GridAPI/ConditionalFormat.ts (hunk 4 above)
Comment (OS-giulianasilva) [reply src/OutSystems/GridAPI/ConditionalFormat.ts]: Fixed

**Review by JoaoFerreira-FrontEnd — APPROVED** [review]

**Review by BenOsodrac — APPROVED** [review]

**Pull request merged**